src/lib/syncDocsPath.ts

```diff
@@ -127,6 +130,40 @@ async function pushDoc(
     });
 }
 
+function sortFiles(files: string[], { allowedFileExtensions }: { allowedFileExtensions: string[] }): string[] {
+  const extensionRegexp = new RegExp(`(${allowedFileExtensions.join('|')})$`);
+
+  const filesBySlug = files
+    .map(filePath => {
+      const doc = frontMatter(fsSync.readFileSync(filePath, 'utf8'));
+      const slug = path.basename(filePath).replace(extensionRegexp, '');
+      const parentDocSlug = doc.data.parentDocSlug;
+
+      return { filePath, slug, parentDocSlug };
```

Comment:
we have a `readDoc` utility that should take care of this, mind using that instead? also it appears we're reading each file twice now, any way we could optimize that[^1]?

[^1]: just flagging it so we're aware, but i don't think it's a real issue!